### PR TITLE
#3314 - Introducing filters for Groups and Users listing in identity server with SCIM2

### PR DESCRIPTION
**Pull request by shyabithd**
Integration tests added for Group and User filtering with SCIM2 in Identity Server

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2GroupTestCase.java [hunk 1]
```diff
@@ -265,27 +265,13 @@ public void testGetGroup() throws Exception {
...
-        String groupNameFromResponse = ((JSONObject) ((JSONArray) ((JSONObject) responseObj).get("Resources")).get(0))
-                .get(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE).toString();
-        //TODO: groupsName comes from this is <Userstore_domain>/displayName
-        assertTrue(groupNameFromResponse.contains(GROUPNAME));
-
-        String groupId = ((JSONObject) ((JSONArray) ((JSONObject) responseObj).get("Resources")).get(0)).get
-                (ID_ATTRIBUTE).toString();
-        assertEquals(groupId, this.groupId);
+        validateFilteredGroup(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE, "+Eq+", GROUPNAME);
+        validateFilteredGroup(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE, "+Sw+", "scim2");
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2GroupTestCase.java]: shall we define constants for the operators and the corresponding values?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2GroupTestCase.java [hunk 2]
```diff
@@ -331,4 +317,33 @@ private String getUsersPath() {
     private String getAuthzHeader() {
         return "Basic " + Base64.encodeBase64String((adminUsername + ":" + password).getBytes()).trim();
     }
+
+    private void validateFilteredGroup(String attributeName, String operator, String searchAttribute) throws IOException {
+
+        String userResourcePath = getPath() + "?filter=" + attributeName + operator +
+                searchAttribute;
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2GroupTestCase.java]: formatting issue here. we can use searchAttribute also in the above line.

Comment (shyabithd) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2GroupTestCase.java]: fixed

**Comment by thanujalk** [comment]
Please merge this after merging https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/104

**Comment by madurangasiriwardena** [comment]
@thanujalk as wso2-extensions/identity-inbound-provisioning-scim2#104 is already merged, shall we merge this as well?

**Pull request merged**